ros2/src/airsim_ros_pkgs/src/airsim_ros_wrapper.cpp

```diff
@@ -210,13 +225,25 @@ void AirsimROSWrapper::create_ros_pubs_from_settings_json()
                     }
                     // if {DepthPlanar, DepthPerspective,DepthVis, DisparityNormalized}, get float image
                     else {
-                        current_image_request_vec.push_back(ImageRequest(curr_camera_name, curr_image_type, true));
+                        if (capture_setting.image_type >= 0) {
+                            current_image_request_vec.push_back(ImageRequest(curr_camera_name, curr_image_type, true));
+                        }
                     }
 
-                    const std::string camera_topic = topic_prefix + "/" + curr_camera_name + "/" + image_type_int_to_string_map_.at(capture_setting.image_type);
-                    image_pub_vec_.push_back(image_transporter.advertise(camera_topic, 1));
-                    cam_info_pub_vec_.push_back(nh_->create_publisher<sensor_msgs::msg::CameraInfo>(camera_topic + "/camera_info", 10));
-                    camera_info_msg_vec_.push_back(generate_cam_info(curr_camera_name, camera_setting, capture_setting));
+                    if (capture_setting.image_type >= 0) {
+                        const std::string camera_topic = topic_prefix + "/" + curr_camera_name + "/" + image_type_int_to_string_map_.at(capture_setting.image_type);
+                        RCLCPP_INFO(nh_->get_logger(), "camera found topic: %s", camera_topic.c_str());
+                        image_pub_vec_.push_back(image_transporter.advertise(camera_topic, 1));
+                        rclcpp::QoS qos_profile = rclcpp::QoS(rclcpp::QoSInitialization::from_rmw(rmw_qos_profile_default))

```

Comment:
This custom QOS policy should make it so that only the most recent image has priority.  There shouldn't be any attempt to keep a buffer of old messages if it gets backed up, since we only care about the most recent image.  This didn't really have a noticeable effect, but I've left it because it makes the most sense to me according to the QOS documentation.